### PR TITLE
Clay Log Integration

### PR DESCRIPTION
**Pull request by jonwinton**
**Includes**
- Integrated `clay-log`
- Removed references to old logging module
- Old logging module is still exported, but there is a deprecation warning about Amphora 5

### CODE REVIEW

File: lib/render.js [hunk 1]
```diff
@@ -340,11 +342,11 @@ function renderExpressRoute(req, res, next) {
       const diff = process.hrtime(hrStart),
         ms = Math.floor((diff[0] * 1e9 + diff[1]) / 1000000);
 
-      log('info', 'rendered express route:' + (req.hostname + req.baseUrl + req.path) + ' ' + ms + 'ms');
+      log('info', `rendered express route: ${req.hostname + req.baseUrl + req.path} ${ms}ms`);
     }
   }).catch((error) => {
     if (error.name === 'NotFoundError') {
-      log('verbose', 'in', req.uri, error.message);
+      log('error', `in ${req.uri}: ${error.message}`);
```
Comment (dmabuada) [inline lib/render.js]: does error.message tell me it's not found? or is that only in error.name?

File: lib/bootstrap.js [hunk 2]
```diff
@@ -265,10 +269,11 @@ function bootstrapComponents(site) {
 module.exports = function () {
   const sites = siteService.sites();
 
+  log('info', 'Beginning bootstrap');
```
Comment (dmabuada) [inline lib/bootstrap.js]: does this need to be logged? if so, a success message would help too

File: lib/responses.js [hunk 3]
```diff
@@ -289,7 +290,7 @@ function notFound(err, res) {
  */
 function serverError(err, res) {
   // error is required to be logged
-  log('error', err);
+  log('error', err.message);
```
Comment (dmabuada) [inline lib/responses.js]: consistent syntax plz

File: lib/render.js (hunk 1 above)
Comment (jonwinton) [reply lib/render.js]: It will tell you it's not found

File: lib/services/components.js [hunk 4]
```diff
@@ -193,9 +195,9 @@ function put(uri, data, locals) {
       const ms = timer.getMillisecondsSince(startTime);
 
       if (ms > timeoutLimit * 0.5) {
-        log('warn', 'slow put', uri, ms + 'ms');
+        log('warn', `slow put ${uri} ${ms}ms`);
       }
-    }).timeout(timeoutLimit, 'Component module PUT exceeded ' + timeoutLimit + 'ms:' + uri);
+    }).timeout(timeoutLimit, `Component module PUT exceeded ${timeoutLimit}ms: ${uri}`);
```
Comment (dmabuada) [inline lib/services/components.js]: can we log this timeout?

File: lib/bootstrap.js (hunk 2 above)
Comment (jonwinton) [reply lib/bootstrap.js]: Good call, one minute

File: lib/services/notifications.js [hunk 5]
```diff
@@ -30,7 +32,7 @@ function callWebhook(event, url, data) {
   }
 
   return rest.fetch(url, options).then(function (response) {
-    log('info', 'called webhook', _.pick(response, ['status', 'statusText']));
+    log('info', `called webhook ${_.pick(response, ['status', 'statusText'])}`);
```
Comment (dmabuada) [inline lib/services/notifications.js]: do we need to log allllll resp statuses?

File: lib/services/pages.js [hunk 6]
```diff
@@ -291,11 +293,11 @@ function publish(uri, data, locals) {
...
 
       if (ms > timeoutLimit * 0.5) {
-        log('warn', 'slow publish', uri, ms + 'ms');
+        log('warn', `slow publish ${uri} ${ms}ms`);
       } else {
-        log('info', 'published', references.replaceVersion(uri), ms + 'ms');
+        log('info', `published ${references.replaceVersion(uri)} ${ms}ms`);
       }
-    }).timeout(timeoutLimit, 'Page publish exceeded ' + timeoutLimit + 'ms:' + uri);
+    }).timeout(timeoutLimit, `Page publish exceeded ${timeoutLimit}ms: ${uri}`);
```
Comment (dmabuada) [inline lib/services/pages.js]: here too.

File: lib/services/pages.test.js [hunk 7]
```diff
@@ -10,25 +10,26 @@ const _ = require('lodash'),
   notifications = require('./notifications'),
   sinon = require('sinon'),
   siteService = require('./sites'),
-  timer = require('../timer'),
-  winston = require('winston');
+  timer = require('../timer');
```
Comment (dmabuada) [inline lib/services/pages.test.js]: does winston need to stay for backwards compatibility?

File: test/fixtures/api-accepts.js [hunk 8]
```diff
@@ -341,6 +340,8 @@ function stubFiles(sandbox) {
   files.getComponentPath.withArgs('missing').returns('missingThing');
   files.getComponentPath.withArgs('invalid').returns(null);
 
+  sandbox.stub(files, 'getComponents').returns(['clay-c5', 'clay-c3', 'clay-c4']);
```
Comment (dmabuada) [inline test/fixtures/api-accepts.js]: i have no idea what this does...but this file removes logging?

Comment (jonwinton) [reply test/fixtures/api-accepts.js]: No, it just makes the function `getComponents` return this array when called from API tests.

File: lib/services/pages.test.js (hunk 7 above)
Comment (jonwinton) [reply lib/services/pages.test.js]: Yes

File: test/fixtures/api-accepts.js (hunk 8 above)
Comment (jonwinton) [reply test/fixtures/api-accepts.js]: Oh, it removes the stub for the old logger from the test file, not from anything else. API tests are looking for the response objects, not what was logged. Individual files do that.

File: lib/services/notifications.js (hunk 5 above)
Comment (jonwinton) [reply lib/services/notifications.js]: It only logs `status` and `statusText`

Comment (dmabuada) [reply lib/services/notifications.js]: i know, but would it be more worthwhile to log non-200 resps?

**Pull request merged**